### PR TITLE
Build: tsup to build vite-builder

### PR DESCRIPTION
**Pull request by ndelangen**
Related: #18732 

## What I did

Migrated the vite-builder to be build with TSup

### CODE REVIEW

File: code/lib/builder-vite/package.json [hunk 1]
```diff
@@ -42,8 +65,26 @@
     "typescript": "^4.9.3",
     "vite": "^3.1.3"
   },
+  "peerDependencies": {
```
[the remark this answers is not in the document]
Comment (ndelangen) [reply code/lib/builder-vite/package.json:68]: I'd make it a regular dep, myself.
but make the version range very open.

[the remark this answers is not in the document]
Comment (IanVS) [reply code/lib/builder-vite/package.json:68]: Sounds good.  Want to do that in this PR?  I think that `>= 3.0.0` should work.  4.0 is in alpha right now, though, which won't be included in that range and we'll end up with potentially duplicated versions.  I don't know of a way for `dependencies` to handle that situation.  Whereas `peerDependencies` of `>= 3.0.0 || >= 4.0.0-0` would accept either one, I think.  But maybe the pain of peer deps isn't worth it, and if someone wants to use the alpha of vite they just need to be sure to use `overrides` or `resolutions` in their package.json.

**Review by IanVS — CHANGES_REQUESTED** [review]
I don't think we can leave vite in devDependencies or it will be prebundled.

**Comment by ndelangen** [comment]
@IanVS it's in both, therefore not pre-bundled.

**Comment by IanVS** [comment]
Thanks, what is the purpose of keeping it in devDependencies if it is also in dependencies?

**Review by IanVS — APPROVED** [review]

**Pull request merged**